packages/noya-state/src/reducers/applicationReducer.ts

```diff
@@ -195,8 +196,17 @@ export function applicationReducer(
           accessPageLayers(draft, pageIndex, layerIndexPaths).forEach(
             (layer) => {
               if (!layer.style) return;
-
               layer.style = styleReducer(layer.style, action);
+
+              if (
+                action[0] === 'setFillFillType' &&
+                action[2] === Sketch.FillType.Pattern &&
+                layer.style.fills
+              ) {
+                const _ref = layer.style.fills[action[1]].image?._ref;
+                if (!_ref) return;
+                draft.sketch.images[_ref] = CHECKERED_BACKGROUND_BYTES;
```

Comment:
Currently, this will replace the image if the user chooses another fill type, e.g. gradient, and then sets it back to pattern

We might want to do this logic before we call `styleReducer()`. We could even do it separately from the big list of cases (`addNewBorder`, `addNewFill`,...) as a top-level case, rather than calling into the `styleReducer`, if that makes things easier